Assets/Player/PsychosisBar.cs

```diff
@@ -0,0 +1,51 @@
+using UnityEngine;
+using UnityEngine.UI;
+
+public class PsychosisBar : MonoBehaviour
+{
+    [SerializeField] private Slider slider;
+    private float _timer;
+    [SerializeField] float currentValue;
+    [SerializeField] private float _coolDownTime;
+    [SerializeField] private float _coolDownSpeed;
+    [SerializeField] private float _barAdd;
+    // Start is called before the first frame update
+    void Start()
+    {
+        slider.value = currentValue;
+        _barAdd = 1.0f;
+    }
+    // Update is called once per frame
+    void Update()
+    {
+        _timer += Time.deltaTime;
+        if (currentValue >= slider.maxValue)
+        {
+            NoReturnPoint();
+        }
+        //if the time is bigger than cooldown, it decreases the bar value and restarts the cooldown
+        else if (_timer >= _coolDownTime)
+        {
+            currentValue -= _coolDownTime / _coolDownSpeed;
+            currentValue = Mathf.Clamp(currentValue, slider.minValue, slider.maxValue);
```

Comment:
While this works really good right now, keep in mind we won't have a UI progress bar in our game so `slider.minValue` and `slider.maxValue` won't exist. Try to have the value in a percentage (from 0f to 1f) instead as it will prevent issues